src/common/snippets/include/snippets/pass/tokenization_config.hpp

```diff
@@ -5,21 +5,43 @@
 #pragma once
 
 #include <cstddef>
+#include <utility>
 
 #include "openvino/core/except.hpp"
 
 namespace ov::snippets::pass {
 
+/**
+ * @struct MatMulConfig
+ * @brief Configuration for MatMul operation tokenization
+ * @ingroup snippets
+ */
+struct MatMulConfig {
+    // Transpose before MatMul on input 0
+    bool is_supported_transpose_a = true;
+    // Transpose before MatMul on input 1
+    bool is_supported_transpose_b = true;
+    // Transpose after MatMul on its output
+    bool is_supported_transpose_c = true;
```

Comment:
I suppose just boolean values are not enough here: not all types of transposes are actually supported even on X86. Ideally, we need to have lambdas here which will replace `is_supported_transpose` in collapse_subgraph.cpp and `is_valid_transpose` in mha_tokenization.cpp

Reply:
That's a good point to discuss it. Right now, these flags are just being used in `is_supported_transpose`. It seemed for me that this way we will get less code duplication here and the logic seems to look simpler. What do you think?